compiler/rustc_driver/src/lib.rs

```diff
@@ -139,76 +139,13 @@ pub fn diagnostics_registry() -> Registry {
     Registry::new(rustc_error_codes::DIAGNOSTICS)
 }
 
-/// This is the primary entry point for rustc.
-pub struct RunCompiler<'a, 'b> {
-    at_args: &'a [String],
-    callbacks: &'b mut (dyn Callbacks + Send),
-    file_loader: Option<Box<dyn FileLoader + Send + Sync>>,
-    emitter: Option<Box<dyn Write + Send>>,
-    make_codegen_backend:
-        Option<Box<dyn FnOnce(&config::Options) -> Box<dyn CodegenBackend> + Send>>,
-}
-
-impl<'a, 'b> RunCompiler<'a, 'b> {
-    pub fn new(at_args: &'a [String], callbacks: &'b mut (dyn Callbacks + Send)) -> Self {
-        Self { at_args, callbacks, file_loader: None, emitter: None, make_codegen_backend: None }
-    }
-
-    /// Set a custom codegen backend.
-    ///
-    /// Used by cg_clif.
-    pub fn set_make_codegen_backend(
-        &mut self,
-        make_codegen_backend: Option<
-            Box<dyn FnOnce(&config::Options) -> Box<dyn CodegenBackend> + Send>,
-        >,
-    ) -> &mut Self {
-        self.make_codegen_backend = make_codegen_backend;
-        self
-    }
-
-    /// Emit diagnostics to the specified location.
-    ///
-    /// Used by RLS.
-    pub fn set_emitter(&mut self, emitter: Option<Box<dyn Write + Send>>) -> &mut Self {
-        self.emitter = emitter;
-        self
-    }
-
-    /// Load files from sources other than the file system.
-    ///
-    /// Used by RLS.
-    pub fn set_file_loader(
-        &mut self,
-        file_loader: Option<Box<dyn FileLoader + Send + Sync>>,
-    ) -> &mut Self {
-        self.file_loader = file_loader;
-        self
-    }
-
-    /// Parse args and run the compiler.
-    pub fn run(self) -> interface::Result<()> {
-        run_compiler(
-            self.at_args,
-            self.callbacks,
-            self.file_loader,
-            self.emitter,
-            self.make_codegen_backend,
-        )
-    }
-}
-fn run_compiler(
+// Primary entry point used by rustc, clippy, and miri.
+pub fn run_compiler(
     at_args: &[String],
     callbacks: &mut (dyn Callbacks + Send),
-    file_loader: Option<Box<dyn FileLoader + Send + Sync>>,
-    emitter: Option<Box<dyn Write + Send>>,
-    make_codegen_backend: Option<
-        Box<dyn FnOnce(&config::Options) -> Box<dyn CodegenBackend> + Send>,
-    >,
```

Comment:
I still have a use case for both `file_loader` (using it for hooking rust-analyzers VFS into rustc at some point for running rustc without having to save) and `make_codegen_backend` (setting the codegen backend from a custom driver where the custom codegen backend has arbitrary data. used by the hotswapping branch of cg_clif)

Reply:
Argh. This is frustrating when there is no way to tell if an interface like this is being used.

Reply:
Yeah